main/tests.py

```diff
@@ -1 +1,73 @@
-# Create your tests here.
+import pytest
+from dogowner.models import DogOwner
+
+
+@pytest.fixture
+def create_dog_owner_user():
+    return DogOwner.create(email='dogowner@address.com',
+                           username='dogOwnerUser01',
+                           password='password123',
+                           dog_name='dog name',
+                           first_name='test',
+                           last_name='user',
+                           phone_number=1234567890,
+                           dog_race='dog race',
+                           dog_picture_url='https://www.google.com/user1.jpg',
+                           dog_age=4,
+                           dog_weight=2,
+                           dog_gender='M'
+                           )
+
+
+@pytest.mark.django_db
+class TestLoginView:
+
+    def test_enter_login_page(self, client):
+        response = client.get('/login/')
+        assert response.status_code == 200
+
+    def test_valid_login_dog_owner_user_info(self, client, create_dog_owner_user):
+        prev_logged_user = client.get('/').wsgi_request.user
+        form = {'username': 'dogOwnerUser01',
+                'password': 'password123',
+                }
+
+        response = client.post('/login/', form, follow=True)
+        curr_log_user = response.wsgi_request.user
+        assert curr_log_user == create_dog_owner_user.user
+        assert prev_logged_user != curr_log_user
+
+    def test_invalid_login_dog_owner_user_info(self, client, create_dog_owner_user):
+        forms = {'username': "daycare@address.com",
+                 'password': "incorrect",
+                 }
+        response = client.post('/login/', forms, follow=True)
+        assert create_dog_owner_user.user != response.wsgi_request.user
+        assert b"Please enter a correct username and password." in response.content
+
+
+@pytest.mark.django_db
+class TestLogoutView:
+    def test_successful_logout(self, client, create_dog_owner_user):
+        client.force_login(user=create_dog_owner_user.user)
+        logged_user = client.get('/').wsgi_request.user
+        response = client.get('/logout/')
+        assert response['Location'] == '/login/'
+        assert response.wsgi_request.user != logged_user
+
+
+@pytest.mark.django_db
+class TestIndexView:
+
+    def test_root_entrypoint_redirection_unlogged_user(self, client):
+        response = client.get("/")
```

Comment:
what about pages that are not the entrypoint? say the user type the URL to access it directly?

worth a test IMO.

Reply:
For now we have only homepage and login pages , and homepage is under @login_required decorator,
do you think I should test this funconallity ?